vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
+    return entries
+
+
+def print_err(msg: str):
+    print(ansi("redf") + msg + ansi("reset"), file=sys.stderr)
+
+
+def read_repos_from_manifest_file(manifest_file: str) -> Dict[str, Dict[str, Any]]:
     try:
-        repos = get_repositories(args.input)
-    except RuntimeError as e:
-        print(ansi('redf') + str(e) + ansi('reset'), file=sys.stderr)
-        return 1
+        with open(manifest_file, "r", encoding="utf-8") as infile:
+            return get_repositories(infile)
+    except RuntimeError as ex:
+        print_err(str(ex))
+    return None
+
 
-    # Get clients from the client directory
-    command = CompareCommand(args)
-    clients = find_repositories(command.paths, nested=command.nested)
-    jobs = generate_jobs(clients, command)
-    result = execute_jobs(
-        jobs, show_progress=True, number_of_workers=args.workers,
-        debug_jobs=args.debug)
+def filter_compare_output_per_path(
+    results: List[Dict[str, Any]], root_path: str
+) -> Dict[str, CompareOutput]:
+    """Removes entries in results which failed and extracts the CompareOutput from the 'output'
+    attribute, returning a dict of output_per_path."""
+    output_per_path: Dict[str, CompareOutput] = {}
+    for result in results:
+        path = result["cwd"].replace(f"{root_path}/", "")
+        output = result["output"]
+        if result["returncode"] != 0:
+            print_err(f"Compare command failed for repo {path}: {output}")
+            continue
+        assert isinstance(output, CompareOutput)
+        output_per_path[path] = output
+    return output_per_path
 
-    table = PrettyTable()
-    table.field_names = ['S', 'Repository', 'Client', 'Version']
 
-    for key in repos:
-        table.add_row(['x', key, repos[key]
-                      ['type'], repos[key]['version']])
+def main(args=None, stdout=None, stderr=None) -> None:
+    set_streams(stdout=stdout, stderr=stderr)
 
-    for entry in result:
-        # Strip the input path from the client path
-        path = entry['cwd'].replace(f'{args.path}/', '')
-        output = entry['output']
-    table.sortby = 'Repository'
-    print(table)
+    parser = CompareCommand.get_parser()
+    add_common_arguments(
+        parser, skip_hide_empty=True, skip_nested=True, skip_repos=True, path_nargs="?"
+    )
+    args = parser.parse_args(args)
+
+    manifest_per_path = read_repos_from_manifest_file(manifest_file=args.input)
+    if manifest_per_path is None:
+        return 1
+
+    result = CompareCommand(args).execute()
+    compare_output_per_path = filter_compare_output_per_path(
+        result, root_path=args.path
+    )
+    if len(compare_output_per_path) == 0:
+        return 1
 
+    entries = generate_table_entries(
+        compare_output_per_path,
+        manifest_per_path,
+        significant_only=args.significant,
+    )
+    print(CompareTable(entries, manifest_file=args.input))
     return 0
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```

Comment:
Nit: may want to leave single quotes for consistency with other commands

Reply:
Ditto: adding pyproject.toml